opentelemetry-api/src/opentelemetry/_logs/_internal/__init__.py

```diff
@@ -0,0 +1,234 @@
+turn is used to create :class:`.Event` and :class:`.Log` objects.
+
+This module provides abstract (i.e. unimplemented) classes required for
+logging, and a concrete no-op implementation :class:`.NoOpLogger` that allows applications
+to use the API package alone without a supporting implementation.
+
+To get a logger, you need to provide the package name from which you are
+calling the logging APIs to OpenTelemetry by calling `LoggerProvider.get_logger`
+with the calling module name and the version of your package.
+
+The following code shows how to obtain a logger using the global :class:`.LoggerProvider`::
+
+    from opentelemetry._logs import get_logger
+
+    logger = get_logger("example-logger")
+
+.. versionadded:: 1.15.0
+"""
+
+import os
+from abc import ABC, abstractmethod
+from logging import getLogger
+from typing import Any, Optional, cast
+
+from opentelemetry._logs.severity import SeverityNumber
+from opentelemetry.environment_variables import _OTEL_PYTHON_LOGGER_PROVIDER
+from opentelemetry.trace.span import TraceFlags
+from opentelemetry.util._once import Once
+from opentelemetry.util._providers import _load_provider
+from opentelemetry.util.types import Attributes
+
+_logger = getLogger(__name__)
+
+
+class LogRecord(ABC):
+    """A LogRecord instance represents an event being logged.
+
+    LogRecord instances are created and emitted via `Logger`
+    every time something is logged. They contain all the information
+    pertinent to the event being logged.
+    """
+
+    def __init__(
+        self,
+        timestamp: Optional[int] = None,
+        observed_timestamp: Optional[int] = None,
+        trace_id: Optional[int] = None,
+        span_id: Optional[int] = None,
+        trace_flags: Optional["TraceFlags"] = None,
+        severity_text: Optional[str] = None,
+        severity_number: Optional[SeverityNumber] = None,
+        body: Optional[Any] = None,
+        attributes: Optional["Attributes"] = None,
+    ):
+        self.timestamp = timestamp
+        self.observed_timestamp = observed_timestamp
+        self.trace_id = trace_id
+        self.span_id = span_id
+        self.trace_flags = trace_flags
+        self.severity_text = severity_text
+        self.severity_number = severity_number
+        self.body = body  # type: ignore
+        self.attributes = attributes
+
+
+class Logger(ABC):
+    """Handles emitting events and logs via `LogRecord`."""
+
+    def __init__(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        schema_url: Optional[str] = None,
+    ) -> None:
+        super().__init__()
+        self._name = name
+        self._version = version
+        self._schema_url = schema_url
+
+    @abstractmethod
+    def emit_event(self, record: "LogRecord") -> None:
+        """Emits a :class:`LogRecord` representing an Event to the processing pipeline."""
+
+    @abstractmethod
+    def emit(self, record: "LogRecord") -> None:
```

Comment:
```suggestion
    def emit_log(self, record: "LogRecord") -> None:
```

Reply:
I will be leaving this as `emit` because there are talks about removing `emit_event` from the logging api.